pkg/collector/corechecks/cluster/orchestrator/terminated_resource_bundle.go

```diff
@@ -103,31 +110,39 @@ func (tb *TerminatedResourceBundle) Stop() {
 
 func toTypedSlice(k8sCollector collectors.K8sCollector, list []interface{}) interface{} {
 	if len(list) == 0 {
-		return nil
+		return list
 	}
 
 	if k8sCollector.Metadata().NodeType == orchestrator.K8sCR || k8sCollector.Metadata().NodeType == orchestrator.K8sCRD {
 		typedList := make([]runtime.Object, 0, len(list))
 		for i := range list {
 			if _, ok := list[i].(runtime.Object); !ok {
-				log.Warn("Failed to convert object to runtime.Object")
+				log.Warnf("failed to cast object to runtime.Object, got type: %T", list[i])
 				continue
 			}
 			typedList = append(typedList, list[i].(runtime.Object))
 		}
 		return typedList
 	}
 
-	// Create a new slice with the type of the object
-	objType := reflect.TypeOf(list[0])
-	typedList := reflect.MakeSlice(reflect.SliceOf(objType), 0, len(list))
+	typedList := reflect.MakeSlice(reflect.SliceOf(k8sCollector.Metadata().ResourceType), 0, len(list))
 
 	for i := range list {
 		typedList = reflect.Append(typedList, reflect.ValueOf(list[i]))
 	}
 	return typedList.Interface()
 }
 
+// checkResourceType checks if the resource is of the correct type
+func checkResourceType(k8sCollector collectors.K8sCollector, resource interface{}) (interface{}, bool) {
```

Comment:
nit: would rename to `getResourceType` since in one case we extract the underlying object from the deletion tombstone. we could also directly return an error instead of boolean